test/InfluxDB.LineProtocol.Tests/Collector/AggregationTests.cs

```diff
@@ -0,0 +1,200 @@
+﻿// ==========================================================================
+// AggregationTests.cs
+// Bus Portal (busliniensuche.de)
+// ==========================================================================
+// All rights reserved.
+// ========================================================================== 
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using InfluxDB.Collector;
+using InfluxDB.Collector.Pipeline;
+using Xunit;
+
+namespace InfluxDB.LineProtocol.Tests.Collector
+{
+    public class AggregationTests
+    {
+        [Fact]
+        public void IncrementsCanBeSummed()
+        {
+            var list = new List<PointData>();
+
+            IPointEmitter collector = new CollectorConfiguration()
+                .Aggregate.AtInterval(TimeSpan.FromMinutes(5))
+                .Aggregate.SumIncrements()
+                .WriteTo.Emitter(pts => list.AddRange(pts))
+                .CreateCollector();
+
+            collector.Emit(new[]
```

Comment:
Does this design require memory space for all of the points in an interval until they're aggregated at emit time? Just clarifying my quick initial reading of the code ... If so it might need a re-think; one of the goals of aggregation, at the client side, would be to maintain a low bound on memory usage in the presence of vast numbers of points. Apologies if I'm misunderstanding, here!

Reply:
You're right. Currently, the aggregation happens at emit time, i.e. if you're using the batcher, it happens when the batcher emits. My goal with this implementation was to reduce the amount of sent data. But your point is totally valid. Reducing the memory footprint would be a nice goal, too.

What is the scenario you're thinking about, i.e. what would be the batch and aggregation timespans?

Reply:
One aspect we should keep in mind is not to accidentally delay the sending of the measurements twice be buffering the aggregation and then buffering the sending.

Reply:
Thanks for the follow-up!

Just seems like a high-allocation metrics implementation is not ideal - generally apps that collect a lot of metrics need to be wary of this kind of caveat. It's not unusual to collect metrics in the multiple-thousands-of-hits/second, so if we're going to try optimizing for this case, aggregating at the point of collection seems pretty appealing (c.f. other metrics implementations).